addons/mail/static/src/js/systray/systray_activity_menu.js

```diff
@@ -111,11 +101,11 @@ const ActivityMenu = Widget.extend({
             if (targetAction.data('domain')) {
                 domain = domain.concat(targetAction.data('domain'));
             }
-
+            const activityGroup = this.activityMenuView.messaging.models['ActivityGroup'].get(ev.target.closest('.o_systray_activity').dataset['activityGroupLocalId']);
             this.do_action({
                 type: 'ir.actions.act_window',
                 name: targetAction.data('model_name'),
-                views: [[false, 'activity'], [false, 'kanban'], [false, 'list'], [false, 'form']],
+                views: [...new Set(['activity'].concat(activityGroup.irModel.availableWebViews))].map(viewName => [false, viewName]),
```

Comment:
order was:
`activity, kanban, list, form`
now is:
`kanban, list, form, activity`

I think order is important

Reply:
Previous code was relying on first item on the list to be the one displayed by default, new code (not sure which commit) is actually specifying the item to display the correct way, so that order no long needs to be changed to display the correct view and can stay consistent.